src/Core/Mesh/Wrapper/TopologicalMeshConvert.cpp

```diff
@@ -0,0 +1,121 @@
+#include <Core/Mesh/Wrapper/TopologicalMeshConvert.hpp>
+#include <Core/Log/Log.hpp>
+
+#include <unordered_map>
+
+namespace Ra {
+namespace Core {
+
+    void MeshConverter::convert(TopologicalMesh& in, TriangleMesh& out){
+
+        struct vertexData{
+            Vector3 _vertex;
+            Vector3 _normal;
+        };
+
+        struct comp_vec{
+            bool operator()(const vertexData &lhv, const vertexData &rhv) const {
+                if( lhv._vertex[0]<rhv._vertex[0] || (lhv._vertex[0]==rhv._vertex[0] &&
+                    lhv._vertex[1]<rhv._vertex[1] ) || (lhv._vertex[0]==rhv._vertex[0] &&
+                    lhv._vertex[1]==rhv._vertex[1] && lhv._vertex[2]<rhv._vertex[2]))
+                {
+                    return true;
+                }
+                return false;
+            }
+        };
+
+        out.clear();
+        typedef std::map<vertexData, int, comp_vec> vMap;
+
+        vMap vertexHandles;
+
+        in.request_face_normals();
+        in.request_vertex_normals();
+        in.update_vertex_normals();
+
+        // iterator over all faces
+        unsigned int vertexIndex = 0;
+        // Use skip iterators
+        for (TopologicalMesh::FaceIter f_it=in.faces_sbegin(); f_it!=in.faces_end(); ++f_it){
+            vertexData v;
+            int indices[3];
+            int i=0;
+            // iterator over vertex (thru halfedge to get access to halfedge normals)
+            for(TopologicalMesh::FaceHalfedgeIter fv_it = in.fh_iter(*f_it); fv_it.is_valid(); ++fv_it){
+                assert(i<3);
+                TopologicalMesh::Point p = in.point(in.to_vertex_handle(*fv_it));
+                TopologicalMesh::Normal n = in.normal(in.to_vertex_handle(*fv_it));
+                v._vertex = Core::Vector3(p[0], p[1], p[2]);
+                v._normal = Core::Vector3(n[0], n[1], n[2]);
+
+                int vi;
+                vMap::iterator vtr = vertexHandles.find(v);
+                if(vtr == vertexHandles.end()){
+                    vi = vertexIndex++;
+                    vertexHandles.insert( vtr, vMap::value_type(v, vi));
+                    //meshVertices.push_back(v);
+                    out.m_vertices.push_back(v._vertex);
+                    out.m_normals.push_back(v._normal);
```

Comment:
Is it possible to allocate beforehand and modify here? would be faster

Reply:
Not sure to catch the point here, could you commit what you want ?

Reply:
Just adding out.m_vertices.reserve(....) before the loop should be ok

Reply:
ok, but there the size I can have is a upper bound (since I do not know the final number of vertices at the end).
Also remember this procedure will eventually be rewritten soon ;)